src/Migration/Service/MigrationDataWriter.php

```diff
@@ -194,6 +195,7 @@ private function handleWriteException(
                     ->withExceptionTrace($exception->getTrace())
                     ->withEntityName($entityName)
                     ->withConvertedData($entity)
+                    ->withEntityId($entity['id'] ?? null)
```

Comment:
Because you changed the builder method to only accept a `string` this shouldn't work anymore. But to be honest I prefer this way of setting it instead of wrapping the call to `->withEntityId` in an `if` branch. I leave it up to you